### PR TITLE
showing the key price for ERC20 locks

### PR DESCRIPTION
**Pull request by julien51**
# Description

This shows the price for ERC20 locks, but requires https://github.com/unlock-protocol/unlock/pull/3686 to be merged first
This pull requests ends up touching a lot of files for 2 big reason:
1. Our "locks" are not very dry (they end up with a lot of shared code... which we should refactor in another PR)
2. Since the locks have to use the `config`, I have had to change a lot of stories to wrap the rendered components inside of a `<ConfigProvider>`. (We should probaby do that at the root storybook config in another PR too)

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/src/components/Paywall.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function Paywall({
   const window = useWindow()
   useEffect(() => {
     postMessage(POST_MESSAGE_READY)
-  }, []) // only send this once, on startup
+  }, [postMessage]) // only send this once, on startup
```
Comment (cellog) [inline paywall/src/components/Paywall.js]: thanks!

File: paywall/src/components/lock/PendingKeyLock.js [hunk 2]
```diff
@@ -11,33 +11,48 @@ import {
...
-      />
-      <TransactionStatus>Waiting for mining confirmation.</TransactionStatus>
-      <Footer>Payment Sent</Footer>
-    </Body>
-  </LockWrapper>
-)
+export const PendingKeyLock = ({ lock, config }) => {
+  const convertCurrency = !lock.currencyContractAddress
+  let currency = 'Eth'
+  if (lock.currencyContractAddress === config.erc20Contract.address) {
```
Comment (cellog) [inline paywall/src/components/lock/PendingKeyLock.js]: this can be extracted into a util

Comment (julien51) [reply paywall/src/components/lock/PendingKeyLock.js]: That's a good point.

**Comment by julien51** [comment]
image

image

image

image

image

File: paywall/src/components/lock/PendingKeyLock.js (hunk 2 above)
Comment (julien51) [reply paywall/src/components/lock/PendingKeyLock.js]: Added!

**Pull request merged**